File: 03.programmers/lv02/02.Python/cache.py
def solution(cacheSize, cities):
    answer = 0
    cache = []
    
    if cacheSize == 0: 
        answer = 5 * len(cities)
        return answer
    for city in cities:
        c = city.lower()
        if c in cache:
            cache.pop(cache.index(c))
            cache.append(c)
            answer +=1
        else:
            if len(cache) < cacheSize:
                cache.append(c)
            else:
                cache.pop(0)
                cache.append(c)
            answer +=5
    return answer


input_cacheSize = 3
input_cities = ["Jeju", "Pangyo", "Seoul", "NewYork", "LA", "Jeju", "Pangyo", "Seoul", "NewYork", "LA"]
print(solution(input_cacheSize, input_cities))

File: ~2023/03.programmers/lv02/01.C++/cache.cpp
#include<iostream>
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(int cacheSize, vector<string> cities) {
    int answer = 0;
    int size = cities.size();
    vector<string> cache;
    if(!cacheSize)return answer = 5*size;
    for(int i=0; i<cities.size(); i++){
        string tmp = "";
        for(char a : cities[i]) tmp+=tolower(a);
        auto itr = find(cache.begin(), cache.end(),tmp);
        if(itr != cache.end()){
            cache.erase(itr);
            cache.push_back(tmp);
            answer++;
        }
        else{
            if(cache.size() < cacheSize) cache.push_back(tmp);
            else{
                cache.erase(cache.begin());
                cache.push_back(tmp);
            }
            answer +=5;
        }
    }
    return answer;
}
int main(){
    int cacheSize = 3;
    vector<string> cities = {"Jeju", "Pangyo", "Seoul", "NewYork", "LA", "Jeju", "Pangyo", "Seoul", "NewYork", "LA"};
    cout<<solution(cacheSize, cities)<<endl;
    return 0;
}